ccf_transaction_processor/Readme.md

```diff
@@ -6,37 +6,142 @@ https://creativecommons.org/licenses/by/4.0/
+           The modified command looks like `sudo -E add-apt-repository ppa:ansible/ansible -y`. Add a `sudo`
+           prefix to the `ansible-playbook ccf-dependencies-no-driver.yml` command appearing in the same file.
+
+	B. In ccf-dependencies-no-driver.yml (present under the same folder), add the environment option for proxies
+           (see https://docs.ansible.com/ansible/latest/user_guide/playbooks_environment.html for reference)
+
+                environment:
+                    http_proxy: <specify-http-proxy-here>
+                    https_proxy: <specify-https-proxy-here>
+
+	C. Add the repo used by ansible scripts for installing python 3.7
+
+		```bash
+		sudo add-apt-repository ppa:deadsnakes/ppa
+		```
+
+#Build CCF & PDO-TP
+
+CCF uses a combination of cmake& ninja to build the applcaition. Execute the following steps to complete the build process:
+
+        A. Add the following lines to the CMakelists.txt found at $PDO_SRC/ccf_transaction_processor/CCF.
+           (look for other add_ccf_app in this file, add just above or below this)
+
+                add_ccf_app(
+		   pdoenc SRCS ../transaction_processor/pdo_tp.cpp
+		   ../transaction_processor/verify_signatures.cpp
+		)
+
+        B. Create the build folder
+
+                ```bash
+		mkdir $PDO_SRC/ccf_transaction_processor/CCF/build
+		```
+
+        C. Set the build flags
+
+		```bash
+                cd $PDO_SRC/ccf_transaction_processor/CCF/build
+                cmake -GNinja -DCOMPILE_TARGETS=virtual -DBUILD_END_TO_END_TESTS=OFF -DBUILD_SMALLBANK=OFF \
+                              -DBUILD_TESTS=OFF-DBUILD_UNIT_TESTS=OFF ..
+		```
+
+
+        D. Build using ninja
+
+                ```bash
+		ninja
+		```
+
+           If the VM has a small memory (< 4GB), it might be required to reduce the parallelism in the build process using the `-j` flag for
+           the `ninja` command, say ninja -j2 or even ninja -j1 (the default seems to be ninja -j4).
+
+
+If build is successful, the libpdoenc.virtual.so library is created inside the build folder (ignore any other targets that get created)
+
+
+# Deploy CCF with PDO-TP as the application
+
+The following commands start a single node CCF under the virtual enclave mode.  PDO-TP is the hosted application.
+
+```bash
+cd $PDO_SRC/ccf_transaction_processor/CCF/build
+python3.7 -m venv env
+source env/bin/activate
+pip install -q -U -r ../tests/requirements.txt
+python ../tests/start_network.py --gov-script ../src/runtime_config/gov.lua  --label pdo_tp -e virtual --package libpdoenc.virtual.so --node <ip-address:6006>
+```
+
+A single node CCF instance hosting the PDO TP will be available for business use @ ip-address:6006. Here `ip-address` is the address of this VM that can be used to ping this machine from other VMs (this is our use case). Set ip-address to 127.0.0.1 for local testing. If you are behind a proxy, add ip-address to the list of no_proxy bash env variable.
```

Comment:
btw: isn't this what you also will have to define (presumably with http:// prefix?) as `PDO_LEDGER_URL`?  If so, worth mentioning ...

Reply:
Actuallly, might be easiest to define early some env-vars along the lines of 
```
     export PDO_CCF_HOST=172.18.0.1
     export PDO_CCF_PORT=6006
```
and then you can define 
```
export PDO_LEDGER_URL=http://${PDO_CCF_HOST}:${PDO_CCF_PORT}
```
and use `${PDO_CCF_HOST}` and `${PDO_CCF_PORT}` for where you say <ip-address> and 6006?

Reply:
To your first question, yes, sure i will add a note that URL PDO_LEDGER_URL is related (and say how). I would not want the two env variable option, since PDO clients do not see this issue. I would still prefer the admin to do a little more work, and make it easy for the clients.

Reply:
well, these environment variables should make it _easier_ than right now as you have a single place where port and ip is defined rather than several?  That certainly what i did as the ip when using docker was not always the same and it was way easier to redefine the env var and have all commands parameterized.  (E.g., i also added a PDO_CCF_BUILD_DIR var which made commands shorter and also enabled me more easily to have the build dir outside of the git repo (primarily for docker reason as i didn't wanted docker images which drag in 3GB of CCF build artifacts [surprising how big the CCF build dir is ....]